src/shared/services/guideResolverHelper.tsx

```diff
@@ -0,0 +1,147 @@
+import React from 'react'
```

Comment:
As discussed on our call, it's nice that you've moved this to its own file, and while it may be out of scope for this PR, it'd be really nice to also refactor the play command to share this code to get rid of the duplicated logic later on.